### PR TITLE
Code refactoring to improve the SAM score (Part 1)

### PR DESCRIPTION
**Pull request by clover2123**

### CODE REVIEW

**Comment by clover2123** [comment]
These patches improve the SAM score from 3.01 upto 3.27
I also measured the octane score and found that there was no performance effect

File: src/parser/CodeBlock.cpp [hunk 1]
```diff
@@ -167,125 +167,29 @@ InterpretedCodeBlock* InterpretedCodeBlock::createInterpretedCodeBlock(Context*
...
-#endif
+    : InterpretedCodeBlock(ctx, script)
+{
+    m_src = StringView(src, scopeCtx->m_functionStartLOC.index, scopeCtx->m_bodyEndLOC.index);
+    m_parent = parentBlock;
+    m_functionName = scopeCtx->m_functionName;
+    m_functionStart = scopeCtx->m_functionStartLOC;
+    m_functionLength = scopeCtx->m_functionLength;
+    m_parameterCount = scopeCtx->m_parameterCount;
+    m_lexicalBlockIndexFunctionLocatedIn = scopeCtx->m_lexicalBlockIndexFunctionLocatedIn;
```
Comment (bbrto21) [inline src/parser/CodeBlock.cpp:189]: Why not use a member initialization list?

File: src/interpreter/ByteCode.h [hunk 2]
```diff
@@ -2282,14 +2282,14 @@ class ThrowOperation : public ByteCode {
...
-    ThrowStaticErrorOperation(const ByteCodeLOC& loc, char errorKind, const char* errorMessage, AtomicString templateDataString = AtomicString())
+    ThrowStaticErrorOperation(const ByteCodeLOC& loc, uint8_t errorKind, const char* errorMessage, AtomicString templateDataString = AtomicString())
         : ByteCode(Opcode::ThrowStaticErrorOperationOpcode, loc)
         , m_errorKind(errorKind)
         , m_errorMessage(errorMessage)
         , m_templateDataString(templateDataString)
     {
     }
-    char m_errorKind;
+    uint8_t m_errorKind;
```
Comment (bbrto21) [inline src/interpreter/ByteCode.h:2292]: Why don't you use `ErrorCode`?

File: src/interpreter/ByteCodeInterpreter.cpp [hunk 3]
```diff
@@ -1,4 +1,5 @@
 /*
+    static void setObjectPreComputedCaseOperationCacheMiss(ExecutionState& state, Object* obj, const Value& willBeObject, const Value& value, SetObjectPreComputedCase* code, ByteCodeBlock* block);
```
[the remark this answers is not in the document]
Comment (clover2123) [reply src/interpreter/ByteCodeInterpreter.cpp]: Nice catch! Its my mistake. I fixed it :)

File: src/parser/CodeBlock.cpp (hunk 1 above)
Comment (clover2123) [reply src/parser/CodeBlock.cpp:189]: All members are first initialized through the call of constructor `InterpretedCodeBlock(ctx, script)`
```c++
InterpretedCodeBlock::InterpretedCodeBlock(Context* ctx, Script* script, StringView src, ASTScopeContext* scopeCtx, InterpretedCodeBlock* parentBlock, bool isEvalCode, bool isEvalCodeInFunction)
    : InterpretedCodeBlock(ctx, script)
{
    m_src = StringView(src, scopeCtx->m_functionStartLOC.index, scopeCtx->m_bodyEndLOC.index);
```

So, the rest members are re-initialized in the body of the constructor.
I changed this part to remove duplicated member initializations.

File: src/interpreter/ByteCode.h (hunk 2 above)
Comment (clover2123) [reply src/interpreter/ByteCode.h:2292]: I just want to clarify the type and size of member, that's all.
(It's not clear but some compiler may allocate different size for each enum class type)

**Review by bbrto21 — APPROVED** [review]

**Review by ksh8281 — APPROVED** [review]

**Pull request merged**